app/src/main/java/org/sopt/and/feature/signup/SignUpScreen.kt

```diff
@@ -0,0 +1,260 @@
+package org.sopt.and.feature.signup
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.flowWithLifecycle
+import org.sopt.and.R
+import org.sopt.and.core.designsystem.component.SocialLoginButton
+import org.sopt.and.core.designsystem.component.textfield.EmailTextField
+import org.sopt.and.core.designsystem.component.textfield.PasswordTextField
+import org.sopt.and.core.designsystem.theme.ANDANDROIDTheme
+import org.sopt.and.core.extension.toast
+
+@Composable
+fun SignUpRoute(
+    navigateToSignIn: (String, String) -> Unit,
+    viewModel: SignUpViewModel = hiltViewModel(),
+    modifier: Modifier = Modifier,
+) {
+    val signUpState by viewModel.signUpState.collectAsState()
```

Comment:
```suggestion
    val signUpState by viewModel.signUpState.collectAsStateWithLifecycle()
```

생명주기 안정적으로 state 받아와서 사용하는 것을 권장합니다.

Reply:
앗하! 감사합니다!!